packages/hardhat-core/src/internal/core/config/config-validation.ts

```diff
@@ -573,3 +574,17 @@ export function getValidationErrors(config: any): string[] {
   const ioTsErrors = DotPathReporter.report(result);
   return [...errors, ...ioTsErrors];
 }
+
+export function validateResolvedConfig(resolvedConfig: HardhatConfig) {
+  console.log(resolvedConfig.solidity.compilers);
+  const soliditySettings: any[] = resolvedConfig.solidity.compilers.map(
+    (s) => s.settings
+  );
+  const optimiser: any[] = soliditySettings.map((o) => o.optimizer);
+  const runs: number[] = optimiser.map((r): number => r.runs);
+  if (runs[0] > 2 ** 32 - 1) {
+    throw new HardhatError(ERRORS.GENERAL.INVALID_CONFIG, {
```

Comment:
We should use a different error type here. I can do that for you if you want, since it's a bit more involved.

Reply:
@fvictorio I will try from my end and let you know if needed any help